projects/dot-rules/src/lib/rule-action-component.ts

```diff
@@ -69,17 +66,15 @@ export class RuleActionComponent implements AfterViewInit, OnInit {
 
     ngOnInit(): void {
         this.typeDropdown = {
-            options: []
+            options: Object.keys(this.ruleActionTypes).map(key => {
+                const type = this.ruleActionTypes[key];
+                return {
+                    label: type._opt.label,
+                    value: type._opt.value
+                };
+            })
         };
     }
-
-    ngAfterViewInit(): void {
-        Object.keys(this.ruleActionTypes).forEach(key => {
-            const type = this.ruleActionTypes[key];
-            this.typeDropdown.options.push(type._opt);
-        });
-    }
-
     onTypeChange(type: string): void {
```

Comment:
doc

Reply:
None of the rules methods are doc, don't event know where to start, is a whole issue by itself.